test/test_weak.py

```diff
@@ -0,0 +1,570 @@
+# Owner(s): ["module: meta tensors"]
+
+import copy
+import gc
+import random
+import threading
+
+import unittest
+
+import torch
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils.weak import WeakIdKeyDictionary
+
+
+def C():
+    return torch.randn(1)
+
+
+# These tests are ported from cpython/Lib/test/test_weakref.py,
+# but adapted to use tensor rather than object
+class WeakTest(TestCase):
```

Comment:
There is no test in there that ensures that we only hold a weak reference and not a strong one?
Also any test to ensure we don't introduce ref cycles (which would delay deletion in practice)?

Reply:
Weak reference is indirectly tested, as we do test that entries are promptly removed from dictionary when they die (this is how I found _fix_weakref leak).

There were some ref cycle tests in cpython that seemed complicated, but I could try to intro them

Reply:
leaving this as follow up, I think prompt deletion also covers the ref cycle case